src/Map/src/Live/ComponentWithMapTrait.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map\Live;
+
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\Map\Map;
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Attribute\PostMount;
+
+/**
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+trait ComponentWithMapTrait
+{
+    /**
+     * @internal
+     */
+    #[LiveProp(hydrateWith: 'hydrateMap', dehydrateWith: 'dehydrateMap')]
+    #[ExposeInTemplate(getter: 'getMap')]
+    public ?Map $map = null;
+
+    abstract protected function instantiateMap(): Map;
+
+    public function getMap(): Map
+    {
+        if (null === $this->map) {
+            $this->map = $this->instantiateMap();
+        }
+
+        return $this->map;
```

Comment:
```suggestion
        return $this->map ??= $this->instantiateMap();
```

Or is it a matter of PHP compat ?

Reply:
(copy-pasted from ComponentWithFormTrait)